src/core/services/shops.service.ts

```diff
@@ -1,67 +1,134 @@
-import { Injectable } from '@angular/core';
-import { HttpClient, HttpHeaders } from '@angular/common/http';
-import { LatLng, UserLogin, Category } from '../models/Models';
-import { Observable } from 'rxjs';
-import { Shop } from '../models/shop.model';
-import { config } from '../../configs/baseconfig'
+import {Injectable} from '@angular/core';
+import {HttpClient, HttpHeaders} from '@angular/common/http';
+import {LatLng, UserLogin, Category} from '../models/Models';
+import {Observable} from 'rxjs';
+import {Shop} from '../models/shop.model';
+import {config} from '../../configs/baseconfig';
 
 @Injectable({
-  providedIn: 'root'
+  providedIn: 'root',
 })
 export class ShopsService {
-
   private hyderabadLatng: LatLng = {
-    latitude: 17.3850,
-    longitude: 78.4867
-  }
+    latitude: 17.385,
+    longitude: 78.4867,
+  };
+
+  constructor(private http: HttpClient) {}
 
-  constructor(private http: HttpClient) { }
+  addShop(shopObject): Observable<any> {
+    const userDetails: any = JSON.parse(localStorage.getItem('shopsnearbyme'));
```

Comment:
Shops Service should not know anything about how authentication is saved, it should receive it as a parameter.
Maybe including a login service with a function to fetch login details would be a better option. What are your thoughts on this?